### DISCUSSION TITLE
Migration recipe can't run child recipes

### DISCUSSION THREAD

**Discussion by JoshTango** [OrchardCMS/OrchardCore#13155] [General]
I put a recipe in the migrations folder.  I call this recipe from my data migrations class using the Create() method.  In the recipe step I want it to run a child recipe.  I enable the module under features.
If this child recipe is in the migrations folder it will not run and if the child recipe is in the recipe folder it hangs.

**Comment by MikeAlhayek** [comment]
@JoshTango There isn't enough here to try to reproduce it.

I suggest you share with you user setup recipe and your sub recipe so we can see what you are doing.

**Comment by JoshTango** [comment]
I am using the blank theme.
Here is the parent recipe in the migration folder:
```
{
  "name": "Master",
  "displayName": "Master",
  "description": "",
  "author": "",
  "website": "",
  "version": "",
  "issetuprecipe": false,
  "categories": [],
  "tags": [],
  "steps": [
    {
      "name": "recipes",
      "Values": [
        {
          "executionid": "OCPlatform",
          "name": "Test"
        }

      ]
    }
  ]
}
```

**Comment by JoshTango** [comment]
You can add any child recipe you want.

**Comment by MikeAlhayek** [comment]
We'll I can add any valid recipe and it will work sure. But, you are adding one that does not work as per your claim. So I think you should share what you have. The the receipt, file name, location of the file and how you are importing the recipe so we can see if there is missing or if there is something wrong with OC. 

A site note, please format any code/json you share to make it easy for the users to read your posts.

**Comment by JoshTango** [comment]
No I don't think you can call a child recipe from the migrations folder period.  What recipe you use is irrelevant but I will share my code.  I will play around some more.

**Comment by JoshTango** [comment]
`{
  "name": "Test",
  "displayName": "Test",
  "description": "",
  "author": "",
  "website": "",
  "version": "",
  "issetuprecipe": false,
  "categories": [],
  "tags": [],
  "steps": [
    {
      "name": "ContentDefinition",
      "ContentTypes": [
        {
          "Name": "Channel",
          "DisplayName": "Channel",
          "Settings": {
            "ContentTypeSettings": {
              "Creatable": true,
              "Listable": true,
              "Draftable": true,
              "Versionable": true,
              "Securable": true
            },
            "FullTextAspectSettings": {}
          },
          "ContentTypePartDefinitionRecords": [
            {
              "PartName": "Channel",
              "Name": "Channel",
              "Settings": {
                "ContentTypePartSettings": {
                  "Position": "1"
                }
              }
            },
            {
              "PartName": "AliasPart",
              "Name": "AliasPart",
              "Settings": {
                "ContentTypePartSettings": {
                  "Position": "2"
                }
              }
            },
            {
              "PartName": "AutoroutePart",
              "Name": "AutoroutePart",
              "Settings": {
                "ContentTypePartSettings": {
                  "Position": "3"
                }
              }
            },
            {
              "PartName": "CommonPart",
              "Name": "CommonPart",
              "Settings": {
                "ContentTypePartSettings": {
                  "Position": "4"
                }
              }
            },
            {
              "PartName": "HtmlBodyPart",
              "Name": "HtmlBodyPart",
              "Settings": {
                "ContentTypePartSettings": {
                  "Position": "5"
                }
              }
            },
            {
              "PartName": "ListPart",
              "Name": "ListPart",
              "Settings": {
                "ContentTypePartSettings": {
                  "Position": "6"
                }
              }
            },
            {
              "PartName": "TitlePart",
              "Name": "TitlePart",
              "Settings": {
                "ContentTypePartSettings": {
                  "Position": "0"
                }
              }
            }
          ]
        }
      ],
      "ContentParts": [
        {
          "Name": "Channel",
          "Settings": {},
          "ContentPartFieldDefinitionRecords": [
            {
              "FieldName": "HtmlField",
              "Name": "Description",
              "Settings": {
                "ContentPartFieldSettings": {
                  "DisplayName": "Description",
                  "Position": "0"
                }
              }
            }
          ]
        }
      ]
    }
  ]
}`

**Comment by JoshTango** [comment]
Thats my child recipe posted above. 
If its in the migrations folder it won't do anything.  Meaning it wont add a content definition called channel.  If its in the recipe folder well then it hangs.

**Comment by jtkech** [comment]
When you define a migration recipe under a `Migrations` folder, you still need to execute it explicitly by code by using the recipe migrator, for example.

            await _recipeMigrator.ExecuteAsync("exportcontenttodeploymenttarget.recipe.json", this);

You can find this recipe file under the `OC.Contents` module.

**Comment by jtkech** [comment]
Usually you execute such a recipe in a registered `DataMigration`, in the Create() method or in an `UpdateFromX()` method to execute a migration step.

**Comment by JoshTango** [comment]
Yes I have been using that method in c# the whole time.  But the recipe I execute also has a recipe step in the json.  That recipe step is not working if you call the parent from the Create method in the migrations class.

**Comment by JoshTango** [comment]
Here is my c# code in my migrations class:

```        
public async Task<int> CreateAsync()
{
    //return 1;
    await _recipeMigrator.ExecuteAsync("features.json", this);
    await _recipeMigrator.ExecuteAsync("master.recipe.json", this);

    return 1;
}
```

**Comment by JoshTango** [comment]
```
{
  "name": "Master",
  "displayName": "Master",
  "description": "",
  "author": "",
  "website": "",
  "version": "",
  "issetuprecipe": false,
  "categories": [],
  "tags": [],
  "steps": [
    {
      "name": "recipes",
      "Values": [
        {
          "executionid": "OCPlatform",
          "name": "Test"
        }

      ]
    }
  ]
}
```

**Comment by JoshTango** [comment]
I changed the description to clarify I m triggering this parent recipe from the migrations class when I enable the module under features menu option.

**Reply by natelabo** [reply]
I did extensive testing with this a while back and can confirm… calling a child recipe from a parent recipe that is part of a migration did NOT work. In fact it would completely mess up the migration. This feature would be nice as versioning migrations could be as simple as adding child recipes to the master (original version 1) migration recipe.

**Comment by JoshTango** [comment]
@sebastienros see above.
This should be moved back to bug issue.

**Reply by ludovic-th** [reply]
Reproduced. I opened the bug #13914.

**Reply by jtkech** [reply]
@JoshTango @ludovic-th see https://github.com/OrchardCMS/OrchardCore/issues/13914#issuecomment-1610050656 and #13916
